syntax = "proto3";

package order;

service OrderService {
    rpc CreateOrder(CreateOrderRequest) returns (CreateOrderResponse);
    rpc GetOrdersByCafe(GetOrdersRequest) returns (OrdersResponse);
}

message CreateOrderRequest {
    string cafe_id = 1;
    repeated OrderItem items = 2;
}

message OrderItem {
    string item_id = 1;
    int32 quantity = 2;
    double price = 3;
}

message CreateOrderResponse {
    bool success = 1;
    string message = 2;
    string order_id = 3;
    double total_price = 4;
}

message GetOrdersRequest {
    string cafe_id = 1;
}

message Order {
    string order_id = 1;
    string cafe_id = 2;
    double total_price = 3;
    string created_at = 4;
    repeated OrderItem items = 5;
}

message OrdersResponse {
    repeated Order orders = 1;
}

